torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -810,6 +811,15 @@ class AsyncAllreduceWork : public ProcessGroupGloo::AsyncWork {
     GENERATE_ALL_TYPES(dtype, getFunction, fn, op);
     return fn;
   }
+
+
+  std::vector<at::Tensor> result() override {
+    TORCH_CHECK(isCompleted());
```

Comment:
Add an error message to `TORCH_CHECK` here mentioning that work needs to be completed before calling `result()`.